book/integration-service.md

```diff
@@ -0,0 +1,264 @@
+# Integration Service
+
+
+## Overview
+
+The Integration Service is composed of controllers that facilitate automated testing of content produced by the build pipelines. It is mainly responsible for integration testing capabilities.
+
+The Integration service uses the pipeline, snapshot and environment controllers to watch for Component builds, trigger and manage the testing Tekton Pipelines and create releases based on the testing outcome(s).
+
+### Goals
+
+- We need to be able to deploy a specific set of images to an environment and record the results of integration testing.
+- Given a specific set of images, we should be able to tell if they passed application validation tests.
+- We should be able to update one component image at a time in order to support 'slow loop' application validation testing.
+- We should have a capability to automatically promote sets of passed images.
+- To be SLSA compliant, we should also be able to tell what version of the build and test pipelines were used to create each component image.
+
+### High Level Workflow
+
+- When a build pipeline completes, Integration service creates a Snapshot CR representing a new collection of components that should be tested together.
+- When Integration service sees a new Snapshot CR (created either by itself or by a user), it coordinates deployment of the application for testing. It does this by creating new “ephemeral” StoneSoup Environment CRs which trigger the GitOps Service to provision environment.
+- After the environment is ready and the application snapshot has been deployed to it, the integration service tests and validates the application according to user-provided configuration. It does this by executing Tekton Pipelineruns against the ephemeral Environment.
+- When all of the test Tekton PipelineRuns have passed, Integration service triggers a deployment of the application to the lowest known “static” environment defined by the user, usually staging.
+- Finally, if any automatic ReleasePlans have been created by the user in the workspace, it will create a Release CR signalling intent to release the tested content, to be carried out by the [release-service](./release-service.md).
+
```

Comment:
Is there another step before this where the integration service de-provisions the ephemeral test environment? Is the test environment left up and running for some period of time or is it de-provisioned immediately? 
Where do the logs go (especially if the test environment is quickly de-provisioned)?

Reply:
Yes, in the current design the Integration service will de-provision the ephemeral test environment after the testing is done.
We've been looking into providing an option to the users of leaving the environment running for a set amount of time, but that will require a new feature for gracefully handling the cleanup and related actions, both in functional and billing sense.

Reply:
As a quick update, I just got confirmation that the deprovisioning will be fully implemented as part of the [STONE-114](https://issues.redhat.com/browse/STONE-114) feature, so I propose we add the deprovisioning step information to this document as part of that effort.